pkg/syncer/cnsoperator/controller/cnsunregistervolume/cnsunregistervolume_controller.go

```diff
@@ -362,6 +372,86 @@ func (r *ReconcileCnsUnregisterVolume) Reconcile(ctx context.Context,
 	return reconcile.Result{}, nil
 }
 
+// validateVolumeNotInUse validates whether the volume to be unregistered is not in use by
+// either PodVM, TKG cluster or Volume service VM.
+func validateVolumeNotInUse(ctx context.Context, cnsVol cnstypes.CnsVolume, pvcName string,
+	pvcNamespace string, k8sClient clientset.Interface) error {
+
+	log := logger.GetLogger(ctx)
+
+	// Check if the Supervisor volume is not in use by any pods (PodVMs) in the namespace.
+	pods, err := k8sClient.CoreV1().Pods(pvcNamespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		log.Errorf("Failed to list pods in namespace %s with error - %s",
+			pvcNamespace, err.Error())
+		return err
+	}
+
+	for _, pod := range pods.Items {
+		for _, podVol := range pod.Spec.Volumes {
+			if podVol.PersistentVolumeClaim != nil &&
+				podVol.PersistentVolumeClaim.ClaimName == pvcName {
+				log.Warnf("Volume %s is in use by pod %s in namespace %s", cnsVol.VolumeId.Id,
```

Comment:
I see that the caller is also logging the error returned from here.
`log.Warnf` Can be changed to debugf, or return the volume id in the return error statement

Reply:
Makes sense. Converted this and similar log entries in this function to Debug logs, and added volume id in the return error statements.